### DISCUSSION TITLE
Puma error while loading library: libicuuc.so.60

### DISCUSSION THREAD

**Discussion by scytaleqc** [mastodon/mastodon#25806] [Q&A]
I upgraded my server's ubuntu release to 20.04 and am currently getting this error when starting the mastodon-web service:

`bundle[8604]: [8604] ! Unable to load application: LoadError: libicuuc.so.60: cannot open shared object file: No such file or directory - /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/charlock_holmes-0.7.7/lib/charlock_holmes/charlock_holmes.so`

Same with mastodon-sidekiq:
`/home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in require': libicuuc.so.60: cannot open shared object file: No such file or directory - /home/mastodon/live/vendor/bundle/ruby/3.0.0/gems/charlock_holmes-0.7.7/lib/charlock_holmes/charlock_holmes.so (LoadError)
`

The streaming service can start with no errors.

I've tried to reinstall the library:
`sudo apt-get install libicu66
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libicu66 is already the newest version (66.1-2ubuntu2.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.`

It's been suggested i rebuild the library locally but how do you do this?

**Comment by pointybirds** [comment]
I had the same issue after an upgrade to Debian 12 ... the suggestion in this post to run "bundle install --force" fixed it for me: https://github.com/mastodon/mastodon/issues/25784

**Reply by netherwraith** [reply]
Stumbled here because of a malfunctioning web service and the "--force"-flag was my personal saviour here - thanks for the input, don't know why my 4.5.8 stopped working after a simple reboot... 🙏

**Comment by jenbanim** [comment]
`bundle install --force` fixed this for me as well. I see in my unattended upgrade history that a `libicu71` package was updated. I'm well out of my depth here, but is there a chance that this borked the Ruby environment somehow? Should that package be held back? This could be a total red herring as I have no idea what, if any, connection there is between the `libicuuc.so.60` in the error and the `libicu71` in the system packages
